### ISSUE TITLE
1.1.2 yields false positives?

### ISSUE THREAD

**Issue by SHxKM** [microsoft/pyright#485]
**Describe the bug**
At least for me, 1.1.2 is yielding errors that weren't there in the previous version. Example:

<img width="988" alt="Screen Shot 2020-01-13 at 21 03 06" src="https://user-images.githubusercontent.com/6573474/72284022-b165b780-3648-11ea-8b84-1b27f8c635ef.png">

However, `QuerySet` implements `order_by` [as can be seen here](https://docs.djangoproject.com/en/2.2/_modules/django/db/models/query/).

**To Reproduce**
Install 1.1.20.

**Expected behavior**
I *think* it shouldn't be error. If it should, then I'm wondering what's changed since 1.1.19.

**VS Code extension or command-line**
Extenstion

**Comment by erictraut** [comment]
I need additional context to diagnose the problem. Can you provide a more complete code snippet?

**Comment by erictraut** [comment]
From the screen shot, it appears that Pyright is interpreting '_BaseQuerySet' as a string literal, not a type, but I can't tell how your code is using '_BaseQuerySet'.

**Comment by SHxKM** [comment]
> I need additional context to diagnose the problem. Can you provide a more complete code snippet?

What kind of code snippet  would be useful? The function is returning an instance of Django QuerySet:

```python
def get_subscriptions_for_profile(*, profile: UserProfile) -> "QuerySet[Subscription]":
    free_plan_cutoff = config("FREE_PLAN_ARTIST_LIMIT", cast=int)

    if accounts.selectors.profile_is_allowed_access(profile_id=UserProfile.pk):
        profile_subs = Subscription.objects.filter(profile=profile)
    else:
        profile_subs = Subscription.objects.filter(profile=profile).order_by(
            "created_date"
        )[:free_plan_cutoff]
    return profile_subs
```

> From the screen shot, it appears that Pyright is interpreting '_BaseQuerySet' as a string literal, not a type, but I can't tell how your code is using '_BaseQuerySet'.

See the function code and `QuerySet` for implementations. I'm not sure what else I can share..

**Comment by erictraut** [comment]
Pyright thinks that the expression`Subscription.objects.filter(profile=profile)` is returning a string literal, and the `str` class clearly doesn't have a method `order_by`. So we need to figure out why it thinks this expression is returning a `str`.

Some additional questions to help me repro and diagnose the problem:
1. Which module is defining the `Subscription` class? It doesn't appear to be an export of django, so it must be some other package. For example, is it the package "django-subscriptions"?
2. Are you providing type stubs for the package that defines the `Subscription` class? Likewise, are you using type stubs for django? If so, where did you get the type stubs, or did you generate them yourself?
3. If you're not using type stubs, do you have Pyright's "Use Library Code for Types" setting enabled?
4. Which version of django and the other package are you using?

If you hover over the `filter` method, what is the hover text? If you command-click on the `filter` method, where does it take you? Here's what I see when I hover over it. I'm guessing that you're seeing something different.

![Screen Shot 2020-01-13 at 10 24 29 PM](https://user-images.githubusercontent.com/7040122/72319273-9ba9bf80-3653-11ea-8604-1190796c812b.png)

**Comment by layday** [comment]
I'm assuming they are using django-stubs, see https://github.com/typeddjango/django-stubs/blob/master/django-stubs/db/models/query.pyi.

**Comment by erictraut** [comment]
This is a duplicate of #486, which provides a better description of the problem and a reproducible code sample.
